census/views/macros/popovers.html

```diff
@@ -0,0 +1,64 @@
+{% macro popover_comparison_title(dataset, place) -%}
+<strong>{{ dataset.name }}</strong> in <strong>{{ place.name }}</strong>
+{%- endmacro %}
+
+
+{% macro popover_comparison_content(entry, place, dataset, questions, gettext) -%}
+
+{% if not entry %}
+    {{ gettext('There is no data available.') }}
+{% else %}
+<ul class="availability-popover">
+    {% for question in questions -%}
+    {% if question.score %}
+        {% set state = entry.answers[question.id] %}
+        {% if state == 'Yes' %}
+            {% set no = '' %}
+        {% elif state == 'No' %}
+            {% set no = 'not' %}
+        {% else %}
+            {% set no = "unclear if it's" %}
+        {% endif %}
+        <li class="{% if state == 'Yes' %}yes{% elif state == 'No' %}no{% elif state == 'Unsure' %}maybe{% endif %}"><i class="fa fa-{{ question.icon }}"></i>&nbsp;
+            {% if question.id == 'exists' %}
+                {% if state == 'Y' %}
+                    {{ gettext('Data exists') }}
+                {% else %}
+                    {{ gettext('Data does not exist') }}
+                {% endif %}
+            {% elif question.id == 'digital' %}
+                {{ gettext("It's %(no)s digital", no=no) }}
+            {% elif question.id == 'public' %}
+                {{ gettext("It's %(no)s public", no=no) }}
+            {% elif question.id == 'free' %}
+                {{ gettext("It's %(no)s free", no=no) }}
+            {% elif question.id == 'online' %}
+                {{ gettext("It's %(no)s online", no=no) }}
+                {% if no == '' %}
+                    <a href="{{ entry.url }}" title="{{ gettext('%(dataset)s for %(place)s on the web', dataset=dataset.id, place=place.name) }}">{{ gettext('here') }}</a>
+                {% endif %}
+            {% elif question.id == 'machinereadable' %}
+                {{ gettext("It's %(no)s machine readable", no=no) }}
+            {% elif question.id == 'bulk' %}
+                {{ gettext("It's %(no)s available in bulk", no=no) }}
+            {% elif question.id == 'openlicense' %}
+                {{ gettext("It's %(no)s openly licensed", no=no) }}
+            {% elif question.id == 'uptodate' %}
+                {{ gettext("It's %(no)s up to date", no=no) }}
+            {% endif %}
+        </li>
+    {% endif %}
+    {%- endfor %}
```

Comment:
This is nunjucks syntax, rather than JSX. It's a port of the templates used in the Open Data Index codebase (which is Jinja2, I think), so I'm not going to do too much additional work here to optimise it, just enough to get it working in nunjucks.